### PR TITLE
Fully implement Coinbase API error handling #98

### PR DESCRIPTION
**Pull request by einsteinx2**

### CODE REVIEW

File: Balance/Shared/Data Model/APIs/Coinbase/CoinbaseError.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    case invalidScope            = "invalid_scope"
+    case rateLimitExceeded       = "rate_limit_exceeded"
+    
+    // NOTE: Check to see if these actually hit since they use 404 and 500 error codes, we might just get a URLSession error
+    case notFound                = "not_found"
+    case internalServerError     = "internal_server_error"
+    
+    var errorDescription: String? {
+        switch self {
+        case .twoFactorRequired: return ""
```
Comment (reddavis) [inline Balance/Shared/Data Model/APIs/Coinbase/CoinbaseError.swift]: `.twoFactorRequired` returns save value as default. Is this just a placeholder? Other than that LGTM

**Comment by ricburton** [comment]
Nice one :D

Richard

File: Balance/Shared/Data Model/APIs/Coinbase/CoinbaseError.swift (hunk 1 above)
Comment (einsteinx2) [reply Balance/Shared/Data Model/APIs/Coinbase/CoinbaseError.swift]: Whoops looks like I forgot to finish that haha

I'll fix that before merging.

Comment (einsteinx2) [reply Balance/Shared/Data Model/APIs/Coinbase/CoinbaseError.swift]: Good catch

**Pull request merged**